### DISCUSSION TITLE
mkdocs serve works but mkdocs gh-deploy fails silently

### DISCUSSION THREAD

**Discussion by mtillman14** [squidfunk/mkdocs-material#6953] [Q&A] [answered]
When I run `mkdocs serve` it works perfectly. When I run `mkdocs gh-deploy` I first observed the action fail, then I re-ran it and it finished running but the site is not functional, the icons are very large and the site is unreadable and none of the links work (see images).

I tried using `mkdocs gh-deploy -v -s` but I didn't see anything of note there.

Here's the contents of my mkdocs.yml:
site_name: ResearchOS
site_author: Mitchell Tillman
repo_name: ResearchOS
repo_url: https://github.com/ResearchOS/ResearchOS
theme:
  name: material
  features:
    - navigation.path
    - navigation.tabs
    - navigation.tabs.sticky
    - navigation.expand
    - toc.follow
    - navigation.top

plugins:
  - search
  - awesome-pages
  - mkdocstrings:
      default_handler: python
      handlers:
        python:
          rendering:
            show_source: true
          options:
            filters:
              - "!^__.*__$"
              - "!^_"
              - "!^validate"
              - "!^to_json"
              - "!^from_json"
              - "!^load_"
              - "!^save_"

markdown_extensions:
  - pymdownx.highlight:
      use_pygments: true
  - pymdownx.superfences

Correctly rendered site homepage
![Correctly rendered site homepage](https://github.com/squidfunk/mkdocs-material/assets/9288768/5de1810a-e9db-49ca-aa03-0e460fffc080)
Huge icons, rendering incorrectly
![Huge icon](https://github.com/squidfunk/mkdocs-material/assets/9288768/2c24f6f2-0da5-41e6-9bb4-49ad22be6f22)
Weird HTML with broken links
![Weird HTML with broken links](https://github.com/squidfunk/mkdocs-material/assets/9288768/de0b43a1-8098-41a8-bfa0-4f7648f2e454)
GitHub Actions does finish running
![GitHub Action finishes running](https://github.com/squidfunk/mkdocs-material/assets/9288768/56880a9c-09b6-446d-95b8-fc10203d2cc8)

Any thoughts would be appreciated!

**Comment by kamilkrzyskow** [comment, marked as answer]
Hello @mtillman14,
you can always check the source of the website using the `view-source:` prefix: `view-source:https://researchos.github.io/ResearchOS/`
or use the developer tools in your browser to see the accessed links to css etc.

The CSS link on the https://researchos.github.io/ResearchOS/ page points to: https://researchos.github.io/assets/stylesheets/main.7e359304.min.css and it returns 404

The `ResearchOS` part is removed, so your `mkdocs.yml` file lacks a properly configured `site_url` setting:
https://www.mkdocs.org/user-guide/configuration/#site_url

Set it to `https://researchos.github.io/ResearchOS/` and see if all issues will be fixed, I didn't investigate past the CSS part ✌️

**Reply by squidfunk** [reply]
> your `mkdocs.yml` file lacks a properly configured site_url setting:

Correct, this is mandatory for GitHub Pages. Add the following to your `mkdocs.yml`:

``` yaml
site_url: https://researchos.github.io/ResearchOS/
```

**Reply by mtillman14** [reply]
This worked thank you! It's the strangest thing because it was working, then I added some things, then it stopped working.
